bestofjs/bestofjs
# Update monorepo Prettier setup

## Goal

- Upgrade to Prettier v3 + [Sort Imports plugin](https://github.com/ianvs/prettier-plugin-sort-imports#readme)
- Move Prettier deps and setup file at the monorepo level
- Format all files running `format:write` NPM script in each app and package

The plugin setup ensure we show imports in the following order:

- Third-party packages
- Monorepo deps `@repo/*`
- Absolute imports
- Relative imports

#### pretty.config.mjs

```js
/** @type {import('prettier').Config} */
const config = {
  singleQuote: false,
  tabWidth: 2,
  trailingComma: "es5",
  importOrder: [
    "^(react/(.*)$)|^(react$)",
    "^(next/(.*)$)|^(next$)",
    "<THIRD_PARTY_MODULES>",
    "",
    "^@repo/(.*)$", // monorepo dependencies
    "^@/(.*)$",
    "^[./]",
  ],
  importOrderTypeScriptVersion: "5.5.4",
  plugins: ["@ianvs/prettier-plugin-sort-imports"],
};

export default config;
```

## How to test

Run the following command for each app

```sh
pnpm -F bestofjs-nextjs format:check
```
